src/issue/screens/issue.screen.js

```diff
@@ -306,6 +339,63 @@ class Issue extends Component {
       ...new Set([...participantNames, ...contributorNames]),
     ].filter(item => !!item);
 
+    const eventsToRender = fullEvents
+      .filter(({ event }) => event !== 'mentioned' && event !== 'subscribed')
+      .filter(({ event }, index, list) => {
+        if (index === 0) {
+          return true;
+        }
+
+        // Merge events are always followed by a closed event, but we don't
+        // want to render them.
+        if (event === 'closed' && list[index - 1].event === 'merged') {
+          return false;
+        }
+
+        return true;
+      })
+      .reduce((results, event, index, list) => {
+        // Label events are recorded individually, but we want to group them for display
+        const labelEvents = ['labeled', 'unlabeled'];
+        const prevEvent = index > 0 ? list[index - 1] : {};
+
+        if (
+          index > 0 &&
+          labelEvents.includes(event.event) &&
+          labelEvents.includes(prevEvent.event) &&
+          moment(event.created_at).diff(prevEvent.created_at) < 10000 &&
+          event.actor.login === prevEvent.actor.login
+        ) {
+          const labelGroup = results[results.length - 1];
+
+          if (event.event === 'labeled') {
+            labelGroup.labeled.push(event);
+          } else {
+            labelGroup.unlabeled.push(event);
+          }
+        } else if (labelEvents.includes(event.event)) {
+          const labelGroup = {
+            event: 'label-group',
+            labeled: [],
+            unlabeled: [],
+            created_at: event.created_at,
+            actor: event.actor,
+          };
+
+          if (event.event === 'labeled') {
+            labelGroup.labeled.push(event);
+          } else {
+            labelGroup.unlabeled.push(event);
+          }
+
+          results.push(labelGroup);
+        } else {
+          results.push(event);
+        }
+
+        return results;
+      }, []);
```

Comment:
Some of this logic might be able to be moved out into a `src/utils` file so we can test it easier, but it's not the end of the world if that doesn't happen right now.

Reply:
yeah, this would be good to test. i don't see much testing going on currently. am i overlooking it?

Reply:
#407 is where the tests are being added

Reply:
Agreed there's quite a bit of logic here that I would like to see tested (once we have our initial tests added). Definitely something we can do a bit later however.